### PR TITLE
Registration validation improvement

### PR DESCRIPTION
**Pull request by karma-o**
Was done according to [this](https://docs.google.com/document/d/1PS-JSbxMZUMSpK0dnz8WdVK5FOBd4wMETKdaxrzGFUA/edit) docs task
* added point about non-future-proof checks and requirements
* fixed and clarified mock user test data point
* fixed code examples

### CODE REVIEW

**Review by Andriy-dz — APPROVED** [review]
Perfect!

File: java-core/junit/registration-validation.md [hunk 1]
```diff
@@ -4,20 +4,21 @@
...
 * In your tests very important to check **all possible edge cases**. For example, we have min password length 6 characters. 
 In this case would be good to test: empty password (0-length - not-ok), 'abc' (3-length - not-ok), 'abcdf' (5-length - **not-ok edge case**), 
 'abcdef' (6-length - **ok edge case**), 'abcdefgh' (8-length - ok). Few not-ok checks you can have in one not-ok test method. Same for ok-test.
 Testing only 'abc' (3-length - not-ok) and 'abcdefgh' (8-length - ok) is not enough because we do not check how the program behave with edge cases.
 * Make sure that after successful registration user was added to storage and `register` method returned correct user.
-* Use `storageDao.add` instead of `register` method to add user for testing.
+* Use low level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` method to add user for testing. Because we can not use the same method which is under a test to insert some mock test data to our storage, since we can not guarantee its correctness yet.
```
Comment (Nick97-git) [inline java-core/junit/registration-validation.md:13]:
```suggestion
* Use low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` method to add users for testing. Because we can not use the same method which is under a test to insert some mock test data into our storage since we can not guarantee its correctness yet.
```

**Review by Nick97-git — APPROVED** [review]

File: java-core/junit/registration-validation.md [hunk 2]
```diff
@@ -4,20 +4,21 @@
...
 Bad example:
 
 public User register(User user) {
- if (user.getLogin() != null 
-   && user.getPassword() != null
+ if (user.getLogin() == null 
+   && user.getPassword() == null
    && user.getAge() < MIN_AGE) {
    throw new RegistrationException("Invalid data");
  }
```
Comment (hronograf) [inline java-core/junit/registration-validation.md:24]:
```suggestion
 if (user.getLogin() == null 
   || user.getPassword() == null
   || user.getAge() < MIN_AGE) {
   throw new RegistrationException("Invalid data");
 }
```

**Review by KseniiaMakarova — APPROVED** [review]
